app/src/main/java/nodomain/freeyourgadget/gadgetbridge/devices/hplus/HPlusSampleProvider.java

```diff
@@ -0,0 +1,140 @@
+*/
+
+import android.content.Context;
+import android.support.annotation.NonNull;
+
+import java.util.Collections;
+import java.util.List;
+
+import de.greenrobot.dao.AbstractDao;
+import de.greenrobot.dao.query.QueryBuilder;
+import nodomain.freeyourgadget.gadgetbridge.database.DBHelper;
+import nodomain.freeyourgadget.gadgetbridge.devices.AbstractSampleProvider;
+import nodomain.freeyourgadget.gadgetbridge.devices.SampleProvider;
+import nodomain.freeyourgadget.gadgetbridge.entities.DaoSession;
+import nodomain.freeyourgadget.gadgetbridge.entities.Device;
+import nodomain.freeyourgadget.gadgetbridge.entities.HPlusHealthActivitySample;
+import nodomain.freeyourgadget.gadgetbridge.entities.HPlusHealthActivitySampleDao;
+import nodomain.freeyourgadget.gadgetbridge.impl.GBDevice;
+import nodomain.freeyourgadget.gadgetbridge.model.ActivityKind;
+
+public class HPlusSampleProvider extends AbstractSampleProvider<HPlusHealthActivitySample> {
+
+    public static final int TYPE_DEEP_SLEEP = 4;
+    public static final int TYPE_LIGHT_SLEEP = 5;
+    public static final int TYPE_ACTIVITY = -1;
+    public static final int TYPE_UNKNOWN = -1;
+    public static final int TYPE_NONWEAR = 3;
+    public static final int TYPE_CHARGING = 6;
+
+    private GBDevice mDevice;
+    private DaoSession mSession;
+
+    public HPlusSampleProvider(GBDevice device, DaoSession session) {
+        super(device, session);
+
+        mSession = session;
+        mDevice = device;;
+    }
+
+    public int getID() {
+        return SampleProvider.PROVIDER_HPLUS;
+    }
+
+    public int normalizeType(int rawType) {
+        switch (rawType) {
+            case TYPE_DEEP_SLEEP:
+                return ActivityKind.TYPE_DEEP_SLEEP;
+            case TYPE_LIGHT_SLEEP:
+                return ActivityKind.TYPE_LIGHT_SLEEP;
+            case TYPE_ACTIVITY:
+                return ActivityKind.TYPE_ACTIVITY;
+            case TYPE_NONWEAR:
+                return ActivityKind.TYPE_NOT_WORN;
+            case TYPE_CHARGING:
+                return ActivityKind.TYPE_NOT_WORN; //I believe it's a safe assumption
+            default:
+//            case TYPE_UNKNOWN: // fall through
+                return ActivityKind.TYPE_UNKNOWN;
+        }
+    }
+
+    public int toRawActivityKind(int activityKind) {
+        switch (activityKind) {
+            case ActivityKind.TYPE_ACTIVITY:
+                return TYPE_ACTIVITY;
+            case ActivityKind.TYPE_DEEP_SLEEP:
+                return TYPE_DEEP_SLEEP;
+            case ActivityKind.TYPE_LIGHT_SLEEP:
+                return TYPE_LIGHT_SLEEP;
+            case ActivityKind.TYPE_NOT_WORN:
+                return TYPE_NONWEAR;
+            case ActivityKind.TYPE_UNKNOWN: // fall through
+            default:
+                return TYPE_UNKNOWN;
+        }
+    }
+
+
+    @Override
+    public List<HPlusHealthActivitySample> getAllActivitySamples(int timestamp_from, int timestamp_to) {
```

Comment:
I think here's something that can be improved. Could it be that you didn't get any samples with super.getAllActivitySamples() because of the (possibly wrong) activity-kind constants/mapping above?

If the constants and the mapping with normalizeType() and toRawActivityKind() works, you shouldn't need to reimplement the query.